public final class kotlinx.coroutines.flow.internal.FlowExceptions_commonKt extends java.lang.Object
{

    public static final int checkIndexOverflow(int)
    {
        int $i0;
        java.lang.ArithmeticException $r0;

        $i0 := @parameter0: int;

        if $i0 < 0 goto label1;

        return $i0;

     label1:
        $r0 = new java.lang.ArithmeticException;

        specialinvoke $r0.<java.lang.ArithmeticException: void <init>(java.lang.String)>("Index overflow has happened");

        throw $r0;
    }

    public static final void checkOwnership(kotlinx.coroutines.flow.internal.AbortFlowException, kotlinx.coroutines.flow.FlowCollector)
    {
        kotlinx.coroutines.flow.internal.AbortFlowException $r0;
        kotlinx.coroutines.flow.FlowCollector $r1, $r2;

        $r0 := @parameter0: kotlinx.coroutines.flow.internal.AbortFlowException;

        $r1 := @parameter1: kotlinx.coroutines.flow.FlowCollector;

        $r2 = virtualinvoke $r0.<kotlinx.coroutines.flow.internal.AbortFlowException: kotlinx.coroutines.flow.FlowCollector getOwner()>();

        if $r2 != $r1 goto label1;

        return;

     label1:
        throw $r0;
    }
}
